aodndata/soop/soop_tmv_nrt.py

```diff
@@ -134,28 +174,44 @@ def preprocess(self):
             raise RuntimeError(
                 "Missing vessel callsign {callsign} from vocabulary'.".format(callsign=SHIP_CODE))
 
-        f = self.file_collection[0]
+        f_txt = self.file_collection.filter_by_attribute_value('extension', '.txt')
+        f_log = self.file_collection.filter_by_attribute_value('extension', '.log')
+        f_nc = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
 
-        if f.extension == '.nc':
+        if len(f_nc):
+            f_nc = f_nc[0]
             # case where we re-push an existing netcdf file
-            f.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+            f_nc.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
 
-        elif f.extension == '.log':
-            log_filename = os.path.basename(f.src_path)
+        elif len(f_log):
+            f_log = f_log[0]
+            log_filename = os.path.basename(f_log.src_path)
 
             if SOOP_NRT_LOG_PATTERN.match(log_filename) is None:
                 raise InvalidFileNameError(
                     "SOOP NRT input logfile has incorrect naming '{name}'.".format(name=log_filename))
 
             # case to create netcdf files from log files
-            f.publish_type = PipelineFilePublishType.NO_ACTION
-
-            netcdf_filepath = netcdf_writer(f.src_path, self.temp_dir, self.ship_callsign_ls[SHIP_CODE])
+            f_log.publish_type = PipelineFilePublishType.NO_ACTION
```

Comment:
Since we are generating a NetCDF from log files, I think we should archive the original log file now with the same path as on S3

Reply:
the archive is done here actually 
https://github.com/aodn/python-aodndata/pull/117/files/111cac9f339e4a3e97bf7dbd2ed338833c1103e1#diff-ab86673366e9f8fc8689d0db161fade2R208

the data provider only sends zip files. He doesn't send log files at all